lambda/publish-todays-curation/src/curation.test.ts

```diff
@@ -0,0 +1,214 @@
+  it("should return a CurationPath for the given data", ()=>{
+    expect(newCurationPath("region-one","all-recipes", new Date(2024,5,6)))
+      .toEqual({
+        region: "region-one",
+        variant: "all-recipes",
+        year: 2024,
+        month: 6,
+        day: 6
+      });
+  })
+});
+
+describe("curation.validateCurationData", ()=> {
+  beforeEach(() => {
+    s3Mock.reset();
+  });
+
+  it("should return a CurationPath object if the file exists", async () => {
+    s3Mock.on(HeadObjectCommand).resolves({});
+
+    const response = await validateCurationData("some-region", "some-variant", new Date(2024, 2, 3));
+    expect(s3Mock.commandCalls(HeadObjectCommand).length).toEqual(1);
+    const c = s3Mock.commandCalls(HeadObjectCommand)[0];
+    const arg = c.firstArg as HeadObjectCommand;
+    expect(arg.input.Bucket).toEqual("TestBucket");
+    expect(arg.input.Key).toEqual("some-region/some-variant/2024-03-03/curation.json");
+
+    expect(response?.variant).toEqual("some-variant");
+    expect(response?.region).toEqual("some-region");
+    expect(response?.year).toEqual(2024);
+    expect(response?.month).toEqual(3);
+    expect(response?.day).toEqual(3);
+  });
+
+  it("should return null if the file does not exist", async () => {
+    s3Mock.on(HeadObjectCommand).rejects(new NotFound({$metadata: {}, message: ""}))
+
+    const response = await validateCurationData("some-region", "some-variant", new Date(2024, 2, 3));
+    expect(response).toBeNull();
+
+    expect(s3Mock.commandCalls(HeadObjectCommand).length).toEqual(1);
+    const c = s3Mock.commandCalls(HeadObjectCommand)[0];
+    const arg = c.firstArg as HeadObjectCommand;
+    expect(arg.input.Bucket).toEqual("TestBucket");
+    expect(arg.input.Key).toEqual("some-region/some-variant/2024-03-03/curation.json");
+  });
+
+  it("curation.should pass on any other error as an exception", async () => {
+    s3Mock.on(HeadObjectCommand).rejects(new S3ServiceException({
+      $fault: "server",
+      name: "",
+      $metadata: {},
+      message: "Test exception"
+    }))
+
+    await expect(validateCurationData("some-region", "some-variant", new Date(2024, 2, 3))).rejects.toBeInstanceOf(S3ServiceException);
+  });
+});
+
+describe("curation.validateAllCuration", ()=>{
+  beforeEach(()=>{
+    s3Mock.reset();
+    jest.resetAllMocks();
+  });
+
+  it("should return a list of the curation files which do exist", async ()=>{
+    s3Mock.on(HeadObjectCommand).resolvesOnce({})
+      .rejectsOnce(new NotFound({$metadata: {}, message: ""}))
+      .resolvesOnce({})
+      .rejects(new NotFound({$metadata: {}, message: ""}));
+
+    const result = await validateAllCuration(Today, false);
+
+    expect(s3Mock.commandCalls(HeadObjectCommand).length).toEqual(4);
+    for(let i=0; i<4; i++) {
+      const req = s3Mock.commandCalls(HeadObjectCommand)[i].firstArg.input as HeadObjectCommandInput;
+      expect(req.Bucket).toEqual(Bucket);
+      switch(i) {
```

Comment:
definitely non-blocking: perhaps better expressed by iterating over an array of things we'd like to assert?

Reply:
probably yes 😁